pkg/utils/yaml.go

```diff
@@ -0,0 +1,99 @@
+package utils
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"strings"
+
+	"gopkg.in/yaml.v3"
+)
+
+const (
+	// YAMLDoc type for yaml files
+	YAMLDoc = "yaml"
+)
+
+var (
+	errHighDocumentCount = fmt.Errorf("document count was higher than expected count")
+)
+
+// LoadYAML loads a YAML file. Can return one or more IaC Documents.
+// Besides reading in file data, its main purpose is to determine and store line number and filename metadata
+func LoadYAML(filePath string) ([]*IacDocument, error) {
+	iacDocumentList := make([]*IacDocument, 0)
+
+	// First pass determines line number data
+	{ // Limit the scope for Close()
```

Comment:
This scope brace just makes sure file.Close() (from defer) is called after line 65, and not waiting until the very end of the func. Since we do two file passes, two different open API calls, I wasn't sure if the two opens would conflict in any way. I don't think so, but it seemed cleaner to close first then open again rather than keep the handle open until returning

Reply:
Makes sense! Good one!